### ISSUE TITLE
How to add multiple IPs per container?

### ISSUE THREAD

**Issue by grkvlt** [Metaswitch/calico-docker#87]
This is mentioned in the README, but I copuld not find documentation on the lower level API to use?

> Using calicoctl we currently only support one IP address per container, but more than one is possible if you use the lower level APIs.

What should I be doing here? I would actually like to be able to run `calicoctl container add` multiple times with different interface names and IP addresses if possible in future, but until then I am happy to use a different mechanism from [Clocker](http://github.com/brooklyncentral/clocker/) instead.

**Comment by tomdee** [comment]
Do you really need to add multiple interfaces? A safer approach would be to just add multiple IPs to the existing interface.

To add multiple IPs (with the current code) you would need to add the IPs to the ipv(4|6)_nets array in the endpoint info - https://github.com/Metaswitch/calico-docker/blob/master/docs/etcdStructure.md#json-endpoint-config

And then manually adding the IPs to the interfaces (ip addr add ... in the container)

I would propose wrapping all this up in some calicoctl commands, e.g.
* calicoctl container address show 
* calicoctl container address add <IP>
* calicoctl container address remove <IP>

If you're happy with just adding multiple IPs to a single interface then I can look at adding the calicoctl commands for this in the next few days.

**Comment by tomdee** [comment]
Work in progress code for this is on the multiple-ips branch

**Comment by grkvlt** [comment]
@tomdee thanks, this seems like a reasonable approcach. I have provisionally updated Clocker to add mutiple IPs to the `eth1` interface based in the container now, but haven't touched the JSON data in etcd, I will wait until you have merged the `calicoctl` commands.

**Comment by tomdee** [comment]
Fixed in https://github.com/Metaswitch/calico-docker/commit/7f59c8784e7cd720c4fca8e4899e01f65d21c7c0 and in the new release (0.4.0

**Comment by frostynova** [comment]
i tried the new cmd "calicoctl ipv4 container test_ct add 192.168.77.30", it said:
```
ln: failed to create symbolic link ‘/var/run/netns/5742’: No such file or directory
```

**Comment by tomdee** [comment]
@robbrockbank Could you try to repro (and fix) this?

**Comment by robbrockbank** [comment]
@frostynova I've been trying to reproduce this locally without much success.

Do you happen to know what version of calico-docker you are running and also what commands you ran prior to error?

**Comment by robbrockbank** [comment]
Fixed in #114
